src/patternfly/components/Tabs/tabs.scss

```diff
@@ -112,167 +123,294 @@
+    --pf-c-tabs__button--after--BorderColor: transparent;
+  }
+}
+
+
+// Buttons
+.pf-c-tabs__button,
+.pf-c-tabs__scroll-button {
+  border: 0;
+
+  // Button borders
+  &::before,
+  &::after {
     position: absolute;
     top: 0;
     right: 0;
     bottom: 0;
     left: 0;
-    margin-left: var(--pf-c-tabs__item--BorderWidth);
     content: "";
-    border-top: var(--pf-c-tabs__item--m-current--BorderTopWidth) solid var(--pf-c-tabs__item--hover--Color);
-  }
-
-  // Tab item current state
-  &.pf-m-current .pf-c-tabs__button {
-    color: var(--pf-c-tabs__item--m-current--Color);
-
-    &::before {
-      border-bottom-color: var(--pf-c-tabs__item--BackgroundColor);
-    }
-
-    &::after {
-      position: absolute;
-      top: 0;
-      right: 0;
-      bottom: 0;
-      left: 0;
-      margin-left: var(--pf-c-tabs__item--BorderWidth);
-      content: "";
-      border-top: var(--pf-c-tabs__item--m-current--BorderTopWidth) solid var(--pf-c-tabs__item--m-current--Color);
-    }
-  }
-
-  // Secondary tab states
-  .pf-m-tabs-secondary & .pf-c-tabs__button {
-    // Remove current state and hover state
-    &::before,
-    &::after,
-    &:hover::after {
-      content: none;
-    }
-  }
-
-  // Secondary tab hover state
-  .pf-m-tabs-secondary &:not(.pf-m-current) .pf-c-tabs__button:hover {
-    color: var(--pf-c-tabs__item--hover--Color);
+    border-style: var(--pf-c-tabs__button--BorderStyle);
+    border-width: 0;
   }
 }
 
-
 // Tab button
 .pf-c-tabs__button {
   position: relative;
-  display: inline-block;
+  display: flex;
+  flex: 1;
   padding: var(--pf-c-tabs__button--PaddingTop) var(--pf-c-tabs__button--PaddingRight) var(--pf-c-tabs__button--PaddingBottom) var(--pf-c-tabs__button--PaddingLeft);
   font-weight: var(--pf-c-tabs__button--FontWeight);
   color: var(--pf-c-tabs__button--Color);
   user-select: none;
-  background: var(--pf-c-tabs__button--Background);
+  background-color: var(--pf-c-tabs__button--Background);
   border: 0;
   outline-offset: var(--pf-c-tabs__button--OutlineOffset);
 
-  // Tab button borders
   &::before {
-    position: absolute;
-    top: 0;
-    right: 0;
-    bottom: 0;
-    left: 0;
-    content: "";
-    border: solid var(--pf-c-tabs__item--BorderColor);
-    border-width: var(--pf-c-tabs__item--BorderWidth) 0 var(--pf-c-tabs__item--BorderWidth) var(--pf-c-tabs__item--BorderWidth);
+    border-color: var(--pf-c-tabs__button--before--BorderColor);
+  }
+
+  &::after {
+    border-bottom-color: var(--pf-c-tabs__button--after--BorderColor);
+    border-bottom-width: var(--pf-c-tabs__button--after--BorderWidth);
+  }
+
+  // Tab item hover state
+  &:hover::after {
+    --pf-c-tabs__button--after--BorderColor: var(--pf-c-tabs__button--hover--after--BorderColor);
   }
 
   &.pf-m-hover,
   &:hover {
     text-decoration: none;
   }
-}
 
-.pf-c-tabs__scroll-item {
-  position: absolute;
-  top: 0;
-  bottom: 0;
-  width: var(--pf-c-tabs__scroll-button--Width);
+  > * {
```

Comment:
here should you actually just target the icon class rather than the child? or do we expect to see other elements inside